src/esp/gfx/Drawable.h

```diff
@@ -130,12 +130,13 @@ class Drawable : public Magnum::SceneGraph::Drawable3D {
   /**
    * Set or change this drawable's @ref Magnum::Trade::MaterialData values from passed material.
    * This is only pertinent for material-equipped drawables.
-   * @param material
+   * @param material material values to set
+   * @param reset whether to reset underlying material or to write over it
    */
-  virtual void setMaterialValues(
-      CORRADE_UNUSED const Magnum::Resource<Magnum::Trade::MaterialData,
-                                            Magnum::Trade::MaterialData>&
-          material) {}
+  virtual void setMaterialValues(CORRADE_UNUSED const Magnum::Resource<
+                                     Magnum::Trade::MaterialData,
+                                     Magnum::Trade::MaterialData>& material,
+                                 CORRADE_UNUSED bool reset) {}
```

Comment:
I don't like boolean argument in public APIs, it's unclear what is it supposed to mean unless you go and look at the definiton. E.g., is it "not setting the material values"? Is it a "dry run"? Etc.. 

Can this be done in some other way? As far as I see, in both the drawable implementations it's delegating to a private `setMaterialValuesInternal()` (for which a bool would be fine, as it's a private API). So having something like this -- for example -- would make sense I think:

```cpp
public:
    void setMaterialValues(material) {
        setMaterialValueInternal(Resource material, false);
    }
    void resetMaterialValues(material) {
        setMaterialValueInternal(Resource material, true);
    }
   
private:
    virtual void setMaterialValuesInternal(Resource material, bool reset);
```

Related useful reading: http://www.gotw.ca/publications/mill18.htm

Reply:
Sure, I like that